### ISSUE TITLE
Create Helper methods for TransactionExecutionStatus response

### ISSUE THREAD

**Issue by tabaktoni** [starknet-io/starknet.js#796]
**Is your feature request related to a problem? Please describe.**
It is fairly complicated for a dev to detect what repose type it should get, as it needs to know the parameter-specific diff between response types.

**Describe the solution you'd like**
I imagine a solution to be in two parts, helper methods:
```ts
function isSuccess(response: TransactionExecutionStatus): SuccessfulTransactionReceiptResponse;
function isRejected(response: TransactionExecutionStatus): RejectedTransactionReceiptResponse;
function isReverted(response: TransactionExecutionStatus): RevertedTransactionReceiptResponse;
```
And one combined callback to use internally those 3 methods
```ts
function isResponse(response: TransactionExecutionStatus, onSuccessCallback, onRevertedCallback, onRejectedCallback)
```

**Additional context**
Final user would use it like:
```ts
const response = await account.waitForTransaction(declareAccount.transaction_hash);
isResponse(response, (success) => {
// here success should be type SuccessfulTransactionReceiptResponse
// do what you do with success...
}, (reverted) => {
// here reverted should be type RevertedTransactionReceiptResponse
// do what you do with reverted...
}, (rejected) => {
// here rejected should be type RejectedTransactionReceiptResponse
// do what you do with rejected...
})
```

**Alternative solution**
Feel free to propose better UX solution.

**Comment by PhilippeR26** [comment]
I have started to work on this subject.
Any Rust devs would immediately say : use Rust enums ; it's made for a such case.
So, my idea is to code something that mimic a Rust Enum :
```typescript
const responseTx = TransactionResponse.of(await account.waitForTransaction(declareAccount.transaction_hash));
responseTx.switch({
    success(txR: SuccessfulTransactionReceiptResponse) { console.log("success =", txR) },
    rejected(txR: RejectedTransactionReceiptResponse) { console.log("rejected =", txR) },
    reverted(txR: RevertedTransactionReceiptResponse) { console.log("reverted =", txR) },
});
console.log("status = ", responseTx.status); // string
console.log("Is a success = ", responseTx.isSuccess); //boolean
console.log("Is rejected = ", responseTx.isRejected);
console.log("Is reverted = ", responseTx.isReverted);

```
As In Rust it's mandatory to code all cases, I will try to force the devs to handle the 3 switch cases.

**Comment by PhilippeR26** [comment]
I know how to generate in my tests a Success or a Reverted transaction response, with the response in `provider.getTransactionReceipt()`.
But how to generate a Rejected response in a test?

**Comment by tabaktoni** [comment]
Resolved in Starknet v6.6.6
